### DISCUSSION TITLE
How to get chapter metadata?

### DISCUSSION THREAD

**Discussion by Rexadev** [audiamus/BookLibConnect#69] [Q&A]
.m4b folder doesn't have anything with it but .aax has 2 json files 1 of them has chapter metadata. How to embed that in file and get chapters.

It would be nice also get other metadata like ASIN in source field and writer and narrator

**Comment by audiamus** [comment]
The .m4b file itself comes with the embedded chapter data as injected by Audible when providing the downloadable file. The .aax file is identical to the .m4b file, renamed for further processing with AAX Audio Converter. The extra json files are an extract of the json response when requesting the download link and the voucher via the Audible Web API. The chapters in the json file often have explicit chapter titles, not just numbers, but that's not always the case. To integrate the information from the json files, together with chapter mark adjustements, run [AAX Audio Converter](https://github.com/audiamus/AaxAudioConverter). AAX Audio Converter also lets you adjust some of the other meta tags.

**Reply by Rexadev** [reply]
Should I run AAX Audio Converter on .aax or .aaxc? If .aax, how do I make BookLibConnect not export to .m4b because it doesn't have json i will be using AAX Audio Converter anyway.

**Reply by audiamus** [reply]
Well, the sole purpose of the "Export" functionality in Book Lib Connect is to provide the files suitable for AAX Audio Converter, as explained in the Readme of this project and in the manual. Just enable it in Book Lib Connect and load the exported AAX files into AAX Audio Converter. AAX Audio Converter should find the associated JSON files.

**Reply by Rexadev** [reply]
How to make it not give .m4b files

**Reply by audiamus** [reply]
You don't. The M4B files are the precondition for the AAX export. Please read the manual.
